### PR TITLE
[Backoffice-internal] Add api post service

### PR DESCRIPTION
**Pull request by Krishanx92**
## Purpose
Sample payload
{
  "name": "PizzaShackAPI",
  "context": "pizza",
  "version": "1.0.0",
  "provider": "admin",
  "lifeCycleStatus": "CREATED"
}

curl -k -X POST -H "Content-Type: application/json" -d @data.json "http://localhost:9443/api/am/backoffice/internal/apis"

### CODE REVIEW

File: backoffice/backoffice-internal-domain-service/ballerina/backoffice_internal_dao.bal [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    if db_client is error {
+        return error("Issue while conecting to databse");
+    } else {
+        //Todo: query need to improve
+        sql:ParameterizedQuery query = `INSERT INTO am_api(api_name, api_version,context,api_provider,status,artifact)
+                                  VALUES (${api.name}, ${api.'version}, ${api.context},${api.provider},${api.lifeCycleStatus}, '{}')`;
+        sql:ExecutionResult result = check db_client->execute(query);
+        return result;
+    }
+}
```
Comment (AmaliMatharaarachchi) [inline backoffice/backoffice-internal-domain-service/ballerina/backoffice_internal_dao.bal:24]:
```suggestion
}

```

File: backoffice/backoffice-internal-domain-service/ballerina/backoffice_internal_dao.bal [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    return dbClient;   
+}
+
+public function db_createAPI(API api) returns sql:ExecutionResult | sql:Error{
+    postgresql:Client | error db_client  = getConnection();
+    if db_client is error {
+        return error("Issue while conecting to databse");
+    } else {
+        //Todo: query need to improve
+        sql:ParameterizedQuery query = `INSERT INTO am_api(api_name, api_version,context,api_provider,status,artifact)
```
Comment (AmaliMatharaarachchi) [inline backoffice/backoffice-internal-domain-service/ballerina/backoffice_internal_dao.bal:19]: Isn't it better to maintain queries in a seperate bal file?

Comment (Krishanx92) [reply backoffice/backoffice-internal-domain-service/ballerina/backoffice_internal_dao.bal]: Fix https://github.com/wso2/apk/pull/276/commits/830d6993e6e9f0a867caed16b80c2aadef75a409

File: backoffice/backoffice-internal-domain-service/ballerina/backoffice_internal_dao.bal (hunk 1 above)
Comment (Krishanx92) [reply backoffice/backoffice-internal-domain-service/ballerina/backoffice_internal_dao.bal]: Fix https://github.com/wso2/apk/pull/276/commits/18723404da5ad5f2c7e7369c76e5139a1bca25e9

**Review by tharindu1st — APPROVED** [review]

**Pull request merged**